src/test/java/uk/gov/beis/els/api/common/ApiValuesFromLegislationCategoryValidatorTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.beis.els.api.common;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.beis.els.api.categories.airconditioners.AirConditionersInternetLabelApiForm;
+import uk.gov.beis.els.categories.internetlabelling.model.InternetLabelFormat;
+import uk.gov.beis.els.categories.internetlabelling.model.InternetLabelOrientation;
+import uk.gov.beis.els.model.RatingClass;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ApiValuesFromLegislationCategoryValidatorTest {
+
+  private Validator validator;
+
+  @Before
+  public void setup() {
+    ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+    validator = validatorFactory.getValidator();
+  }
+
+  @Test
+  public void validRatingClass_displayValue_assertNoErrors() {
+    String validRatingClass = RatingClass.D.getDisplayValue();
+    AirConditionersInternetLabelApiForm form = getForm();
+    form.setEfficiencyRating(validRatingClass);
+
+    Set<ConstraintViolation<AirConditionersInternetLabelApiForm>> errors = validator.validate(form);
+
+    assertEquals(0, errors.size());
+  }
+
+  @Test
+  public void invalidRatingClass_displayValue_assertError() {
+    String invalidRatingClass = RatingClass.G.getDisplayValue();
+    AirConditionersInternetLabelApiForm form = getForm();
+    form.setEfficiencyRating(invalidRatingClass);
+
+    Set<ConstraintViolation<AirConditionersInternetLabelApiForm>> errors = validator.validate(form);
+
+    assertEquals(1, errors.size());
```

Comment:
It would be worth checking that the error thrown is actually coming form the `ApiValuesFromLegislationCategory` validator and not one of the others. This is easy with AssertJ, you should be able to do something like:
```   
assertThat(errors)
        .extracting(ConstraintViolation::getMessageTemplate)
        .containsExactly("{uk.gov.beis.els.api.common.ApiValuesFromLegislationCategory.message}");
```
To check that validation error is of the right type. You could match on the error text but seems better to match on the template so the tests don't rely on specific error text.
Same for the other tests

Reply:
thanks
done